src/Domain/Domain/DomainObjectRepositoryBase.cs

```diff
@@ -17,14 +19,23 @@ public abstract class DomainObjectRepositoryBase : IDomainObjectRepository
         private readonly EventSourceConfiguration configuration;
         private static readonly string TraceCategory = typeof(DomainObjectRepositoryBase).FullName;
         private readonly ILogger logger;
+        private readonly IServiceProvider provider;
+        private readonly IStateActivator stateActivator;
+        private readonly IDomainObjectActivator activator;
 
-        protected DomainObjectRepositoryBase(EventSourceConfiguration configuration)
+        protected DomainObjectRepositoryBase(EventSourceConfiguration configuration, IServiceProvider provider)
```

Comment:
Do you need to inject IServiceProvider here? Can't you just inject the dependencies directly?

Reply:
done